server/frontend/src/components/Planets.js

```diff
@@ -7,39 +7,51 @@ import dataLoader from './data-loader-json.js'
 import '../css-react-virtualized/styles.css'; // only needs to be imported once
 import '../css/components/table.css';
 import '../css/index.css';
+import { List } from 'immutable';
+import Immutable from 'immutable';
+import SortDirection from "./SortDirection";
 
 class Planets extends Component {
 
-  // Maps column name to its unit.
-  units = {
-    'Mass': <span>10<sup>24</sup>kg</span>,
-    'Diameter': 'km',
-    'Density': <span>kg/m<sup>3</sup></span>,
-    'Gravity': <span>m/s<sup>2</sup></span>,
-    'Escape Velocity': 'km/s',
-    'Rotation Period': 'hours',
-    'Length of Day': 'hours',
-    'Distance from Sun': <span>10<sup>6</sup> km</span>,
-    'Perihelion': <span>10<sup>6</sup> km</span>,
-    'Aphelion': <span>10<sup>6</sup> km</span>,
-    'Orbital Period': 'days',
-    'Orbital Velocity': 'km/s',
-    'Orbital Inclination': 'degrees',
-    'Orbital Eccentricity': '',
-    'Obliquity to Orbit': 'degrees',
-    'Mean Temperature': 'C',
-    'Surface Pressure': 'bars',
-    'Number of Moons': 'number',
-    'Ring System?': 'Yes/No',
-    'Global Magnetic Field?': 'Yes/No',
-  }
+  constructor(props) {
+    super(props);
+    const sortBy = 'numberOfMoons';
+    const sortDirection = SortDirection.ASC;
+
+    // Maps column name to its unit.
+    const units = {
+      'Mass': <span>10<sup>24</sup>kg</span>,
+      'Diameter': 'km',
+      'Density': <span>kg/m<sup>3</sup></span>,
+      'Gravity': <span>m/s<sup>2</sup></span>,
+      'Escape Velocity': 'km/s',
+      'Rotation Period': 'hours',
+      'Length of Day': 'hours',
+      'Distance from Sun': <span>10<sup>6</sup> km</span>,
+      'Perihelion': <span>10<sup>6</sup> km</span>,
+      'Aphelion': <span>10<sup>6</sup> km</span>,
+      'Orbital Period': 'days',
+      'Orbital Velocity': 'km/s',
+      'Orbital Inclination': 'degrees',
+      'Orbital Eccentricity': '',
+      'Obliquity to Orbit': 'degrees',
+      'Mean Temperature': 'C',
+      'Surface Pressure': 'bars',
+      'Number of Moons': 'number',
+      'Ring System?': 'Yes/No',
+      'Global Magnetic Field?': 'Yes/No',
+    }
 
-  state = {
-    planets: [],
-    selectedPlanet: null,
-    satellites: []
+    this.state = {
+      planets: List(),
```

Comment:
Due to this change, selection of a planet row stopped to work. We need to apply:

```
diff --git a/server/frontend/src/components/Planets.js b/server/frontend/src/components/Planets.js
index 5466eaa..cbeabc4 100644
--- a/server/frontend/src/components/Planets.js
+++ b/server/frontend/src/components/Planets.js
@@ -54,7 +54,7 @@ class Planets extends Component {
   }

   _rowClassName = ({index}) => {
-    if (this.state.selectedPlanet === this.state.planets[index]) {
+    if (this.state.selectedPlanet === this.state.planets.get(index)) {
       return 'selectedRow';
     }
     if (index % 2 === 0) {
```

Reply:
Of course, get(index) instead [index].